### PR TITLE
Add intro to dev guide

### PR DESCRIPTION
**Pull request by henryiii**
Draft based on https://iscinumpy.dev/post/cookie

- [x] The main subject relates to at least one project affiliated to the Scientific Python Ecosystem.
- [x] I have the right to publish the content under BSD 3-Clause License for the code and Creative Common CC-BY-4.0 License for the text.
- [x] Images have been compressed using a tool like `pngquant`.

### CODE REVIEW

**Comment by stefanv** [comment]
This is awesome.

File: content/post/cookie/introduction/index.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+- Improved CLI experience even if using cookiecutter (like no longer requesting
+  current year).
+- Reworked docs template.
+- Support for cookiecutter 2.2 pretty descriptions (added about four hours after
+  cookiecutter 2.2.0 was released).
+- GitLab CI support when not targeting github.com URLs (added by Giordon Stark).
+
+## Repo-review
+
+I wrote an [introduction post](/post/repo-review/) about this one!
```
Comment (stefanv) [inline content/post/cookie/introduction/index.md:106]: This link currently fails the link checker.

Comment (henryiii) [reply content/post/cookie/introduction/index.md:106]: Oops, copy-and past from my blog where it does work.

Do we have a photo we can add?

Comment (jarrodmillman) [reply content/post/cookie/introduction/index.md:106]: Yes, I will get you add  a couple of photos soon (I am at the airport now).

Comment (jarrodmillman) [reply content/post/cookie/introduction/index.md]: @henryiii Would you mind cross posting the repo-review intro post here as well (i.e., `/posts/scientific-python/repo-review/` in this repository). Maybe you could do that in a separate PR, which we could merge and then update this:
```
<!--fix
I wrote an [introduction post](/posts/scientific-python/repo-review/) about this one!
-->
```

If you prefer not cross-posting, then we should fix the above with the URL of the existing post.

File: content/posts/scientific-python/cookie/introduction/index.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+## Using the guide
+
+If you have a guide, I'd like for you to compare it with the Scientific Python
+Development Guide, and see if we are missing anything - bring it to our
+attention, and maybe we can add it. And then you can link to the centerally
+maintained guide instead of manually maintaining a complete custom guide. See
+[scikit-hep/developer][] for an example; many pages now point at this guide now.
+We can also provide org integrations for the cookiecutter, providing some
+customizations when a user targets your org (targeting `scikit-hep` will add a
```
Comment (jarrodmillman) [inline content/posts/scientific-python/cookie/introduction/index.md:166]: Could we do something similar for `scientific-python` here (i.e., add badges)?

Comment (henryiii) [reply content/posts/scientific-python/cookie/introduction/index.md:166]: Yes, sure. That's the idea, we can offer integration if a specific target org is set. Do you have some in mind?

File: content/post/cookie/introduction/index.md (hunk 1 above)
Comment (henryiii) [reply content/post/cookie/introduction/index.md]: I'd need tabs implemented to cross-post. https://gitlab.com/iscinumpy/iscinumpy.gitlab.io/-/blob/master/layouts/shortcodes/tabs.html & https://gitlab.com/iscinumpy/iscinumpy.gitlab.io/-/blob/master/layouts/shortcodes/tab.html.

**Comment by henryiii** [comment]
I've updated this for repo-review 0.10. Any idea when it might be considered ready?  Is there anything left I'm supposed to do?

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
Thanks for the ping, @henryiii; I'm sorry that this dropped off the radar. Let's get it live; we can always make minor tweaks after-the-fact (e.g., if Jarrod wanted to resolve the badge issue).

**Comment by stefanv** [comment]
I notice now that the page lives at https://blog.scientific-python.org/scientific-python/introduction/

This may be a bit broad. Shall we rename `introduction` to `introduction-to-developer-guide` or similar to disambiguate?

**Comment by henryiii** [comment]
Sure!

**Comment by stefanv** [comment]
Please see https://github.com/scientific-python/blog.scientific-python.org/pull/131

**Pull request merged**